oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java

```diff
@@ -153,6 +163,80 @@ public static ImpersonatedCredentials create(GoogleCredentials sourceCredentials
         .build();
   }
 
+  public String getAccount() {
+      return this.targetPrincipal;
+  }
+
+  /**
+   * Signs the provided bytes using the private key associated with the impersonated
+   * service account
+   *
+   * @param toSign bytes to sign
+   * @return signed bytes
+   * @throws SigningException if the attempt to sign the provided bytes failed
+   * @see <a href="https://cloud.google.com/iam/credentials/reference/rest/v1/projects.serviceAccounts/signBlob>Blob Signing</a>
+   */
+  @Override
+  public byte[] sign(byte[] toSign) {
+    BaseEncoding base64 = BaseEncoding.base64();
+    String signature;
+    try {
+      signature = getSignature(base64.encode(toSign));
+    } catch (IOException ex) {
+      throw new SigningException("Failed to sign the provided bytes", ex);
+    }
+    return base64.decode(signature);
+  }
+
+  private String getSignature(String bytes) throws IOException {
+
```

Comment:
nit: extra space

We should have a linting bot running, but I'm not sure why it isn't running.